### PR TITLE
Divyanshu | Test | Tags page, create new button in discount page and border line in field is not as per requirement

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cv9wkp0

### CODE REVIEW

File: apps/web-giddh/src/app/settings/discount/discount.component.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
     (commonLocaleData)="commonLocaleData = $event"
     (translationComplete)="translationComplete($event)"
 >
-    <div class="box clearfix">
+    <div class="box clearfix giddh-container-center">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/discount/discount.component.html]: section-center

File: apps/web-giddh/src/app/settings/tags/tags.component.html [hunk 2]
```diff
@@ -4,120 +4,37 @@
     (localeData)="localeData = $event"
     (commonLocaleData)="commonLocaleData = $event"
 >
-    <div class="box">
-        <form>
+    <div class="box giddh-container-center">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/tags/tags.component.html]: also correct this

File: apps/web-giddh/src/app/settings/tags/tags.component.html [hunk 3]
```diff
@@ -4,120 +4,37 @@
...
+                    <mat-form-field class="w-100" appearance="outline" *ngIf="!isLoading">
+                        <mat-label>{{ localeData?.tags_name  }}</mat-label>
+                        <mat-chip-grid #chipGrid aria-label="Enter tag">
+                            <mat-chip-row *ngFor="let tag of tags"
+                                        (removed)="deleteTag(tag)"
+                                        [editable]="true"
+                                        (edited)="updateTag(tag, $event)"
+                                        [aria-description]="'press enter to edit ' + tag.name"
+                            >
+                            {{tag.name | titlecase }}
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/tags/tags.component.html]:
```suggestion
                            {{tag?.name | titlecase }}
```

File: apps/web-giddh/src/app/settings/tags/tags.component.html [hunk 4]
```diff
@@ -4,120 +4,37 @@
...
+                            <button matChipRemove [attr.aria-label]="'remove ' + tag.name">
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/settings/tags/tags.component.html]:
```suggestion
                            <button matChipRemove [attr.aria-label]="'remove ' + tag?.name">
```

File: apps/web-giddh/src/app/shared/create-address/create-address.component.ts [hunk 5]
```diff
@@ -84,8 +84,11 @@ export class CreateAddressComponent implements OnInit, OnDestroy {
      * @memberof CreateAddressComponent
      */
     public ngOnChanges(changes: SimpleChanges): void {
-        if (changes?.addressConfiguration && changes.addressConfiguration.currentValue !== changes.addressConfiguration.previousValue && (this.addressConfiguration?.stateList?.length || this.addressConfiguration?.countyList?.length)) {
+        if (changes?.addressConfiguration && changes.addressConfiguration.currentValue !== changes.addressConfiguration.previousValue && (this.addressConfiguration?.stateList?.length || this.addressConfiguration?.countyList?.length || this.addressConfiguration?.
+            linkedEntities)) {
             this.setFormData();
+            console.log(this.addressConfiguration);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/create-address/create-address.component.ts]: remove console

File: apps/web-giddh/src/app/shared/header/components/aside-setting/aside-setting.component.html [hunk 6]
```diff
@@ -35,12 +35,22 @@ <h4 class="font-15" *ngIf="showSettingHeading">{{ localeData?.page_heading }}</h
...
+                                <a href="javascript:;" aria-label="icon" class="d-block clearfix" [routerLink]="tab?.link">
+                                    <span [innerHTML]="tab.icon"></span>
+                                    <span>{{ tab.label }}</span>
+                                </a>
+                            </mat-list-item>
+                        </ng-container>
+                        <ng-template #tagsListItem>
+                            <mat-list-item role="listitem" [ngClass]="{ 'active-link': isTagMenuOpened }">
+                                <a href="javascript:;" aria-label="icon" class="d-block clearfix" [matMenuTriggerFor]="tagsMenu" #tagsMenuTrigger="matMenuTrigger" (menuOpened)="isTagMenuOpened = true" (menuClosed)="isTagMenuOpened = false">
+                                    <span [innerHTML]="tab.icon"></span>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/aside-setting/aside-setting.component.html]: ### 
```suggestion
                                    <span [innerHTML]="tab?.icon"></span>
```

File: apps/web-giddh/src/app/shared/header/components/aside-setting/aside-setting.component.html [hunk 7]
```diff
@@ -35,12 +35,22 @@ <h4 class="font-15" *ngIf="showSettingHeading">{{ localeData?.page_heading }}</h
...
+                                    <span>{{ tab.label }}</span>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/shared/header/components/aside-setting/aside-setting.component.html]:
```suggestion
                                    <span>{{ tab?.label }}</span>
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/settings/discount/discount.component.html (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/discount/discount.component.html]: Done

File: apps/web-giddh/src/app/settings/tags/tags.component.html (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/tags/tags.component.html]: Done

File: apps/web-giddh/src/app/settings/tags/tags.component.html (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/tags/tags.component.html]: Done

File: apps/web-giddh/src/app/settings/tags/tags.component.html (hunk 4 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/tags/tags.component.html]: Done

File: apps/web-giddh/src/app/shared/create-address/create-address.component.ts (hunk 5 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/shared/create-address/create-address.component.ts]: Done

File: apps/web-giddh/src/app/shared/header/components/aside-setting/aside-setting.component.html (hunk 6 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/shared/header/components/aside-setting/aside-setting.component.html]: Done

File: apps/web-giddh/src/app/shared/header/components/aside-setting/aside-setting.component.html (hunk 7 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/shared/header/components/aside-setting/aside-setting.component.html]: Done

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/app/settings/tags/tags.component.ts [hunk 8]
```diff
@@ -1,118 +1,141 @@
...
+    /** Holds announcer for mat chip */
+    public announcer: any = inject(LiveAnnouncer);
 
     constructor(
         private settingsTagService: SettingsTagService,
         private toaster: ToasterService,
         public dialog: MatDialog,
-    ) {
-    }
+        private formBuilder: UntypedFormBuilder
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/settings/tags/tags.component.ts]:
```suggestion
        private formBuilder: FormBuilder
```

File: apps/web-giddh/src/app/shared/header/header.module.ts [hunk 9]
```diff
@@ -63,7 +63,8 @@ import { MatToolbarModule } from "@angular/material/toolbar";
         CommandKModule,
         MatDividerModule,
         GenericAsideMenuAccountModule,
-        MatToolbarModule
+        MatTooltipModule,
+        SettingsModule
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/header/header.module.ts]: don't import settings module

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/shared/header/header.module.ts (hunk 9 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/shared/header/header.module.ts]: Done

File: apps/web-giddh/src/app/settings/tags/tags.component.ts (hunk 8 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/settings/tags/tags.component.ts]: Done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**